chore: Set-up the initial codeowners

alexjba opened the pull request:
### What does the PR do

closes #16473

Setting-up the initial code owners in GH. The code owners will be automatically added to PR reviews.
See https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

### Important

Any CC can change the `CODEOWNERS` file to add/remove himself as a default reviewer. `CODEOWNERS` file can work with specific files or folders (E.g. When developing a new feature, the dev can add himself as a default reviewer for the files involved in the feature)

I've took the freedom to add specific CCs to some domain that are clearly defined (CI, e2e, qml). Please feel free to propose changes if you'd like me to extend the ownership or remove your username completely.
CC. @jakubgs @yakimant @micieslak @caybro

@status-im/desktop is the default reviewer
@status-im/desktop-qa is the default reviewer for `/test/e2e/`

<!-- Fill in the relevant information below to help us evaluate your proposed changes. -->

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 9fd1222a | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-16474/1/) | 2024-10-07 08:16:29 | ~6 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-16474/1//consoleText) |

alexjba commented:
Added @status-im/desktop-wallet as default reviewer for wallet code.
https://github.com/status-im/status-desktop/pull/16474/commits/be3c170a3087ca5394c7e3da4d88ed4ce1930bce

status-im-auto commented:
#### :heavy_check_mark: [status-desktop/prs/macos/aarch64/package/PR-16474#2](https://ci.status.im/job/status-desktop/job/prs/job/macos/job/aarch64/job/package/job/PR-16474/2/) :small_blue_diamond: ~4 min 21 sec :small_blue_diamond: be3c170a :small_blue_diamond: [:package: macos/aarch64 package](https://status-im-desktop-prs.ams3.digitaloceanspaces.com/StatusIm-Desktop-241007-083537-be3c17-pr16474-aarch64.dmg)

status-im-auto commented:
#### :heavy_check_mark: [status-desktop/prs/windows/x86_64/package/PR-16474#2](https://ci.status.im/job/status-desktop/job/prs/job/windows/job/x86_64/job/package/job/PR-16474/2/) :small_blue_diamond: ~20 min :small_blue_diamond: be3c170a :small_blue_diamond: [:package: windows/x86_64 package](https://status-im-desktop-prs.ams3.digitaloceanspaces.com/StatusIm-Desktop-241007-083541-be3c17-pr16474-x86_64.exe)

micieslak reviewed (approved):
Cool!

jakubgs commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,22 @@
+# See https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
+
+*           @status-im/desktop                              # By default adding the desktop team to review
+Makefile    @jakubgs @yakimant @status-im/desktop
+/ci/        @jakubgs @yakimant                              # Any change in the ci dir requires approval from Jakub or Anton
+/nix/       @jakubgs @yakimant
```
Why not use use `@status-im/devops` instead of me and Anton explicitly?
Also, we might want to include the `/scripts/` folder as well.

osmaczko reviewed (approved):
:cool:

status-im-auto commented:
#### :heavy_check_mark: [status-desktop/prs/windows/x86_64/package/PR-16474#3](https://ci.status.im/job/status-desktop/job/prs/job/windows/job/x86_64/job/package/job/PR-16474/3/) :small_blue_diamond: ~24 min :small_blue_diamond: 8c5546e3 :small_blue_diamond: [:package: windows/x86_64 package](https://status-im-desktop-prs.ams3.digitaloceanspaces.com/StatusIm-Desktop-241007-134418-8c5546-pr16474-x86_64.exe)

alexjba commented:
Added the devOps team to `Makefile`, `ci`, `nix`, `scripts`.
I've also moved the inline comments. Apparently the inline `#` is an escape character, but on a new line it's a comment.


Looks nice in VS code as well with some plugins
<img width="1148" alt="Screenshot 2024-10-07 at 17 07 15" src="https://github.com/user-attachments/assets/63facd6b-a615-4000-b1c9-2516edba7f45">
